src/pages/api/auth/[...nextauth].js

```diff
@@ -0,0 +1,38 @@
+import NextAuth from 'next-auth';
+import CredentialsProvider from 'next-auth/providers/credentials';
+import GithubProvider from 'next-auth/providers/github';
+
+const providers = [];
+if (process.env.VERCEL_ENV === 'preview') {
+	providers.push(
+		CredentialsProvider({
+			name: 'credentials',
+			credentials: {
```

Comment:
Why do you need this if() statement?
Couldn't you just add the providers without the pushing?
Something like this:
```jsx
export const authOptions = {
  // Configure one or more authentication providers
  providers: [
    GithubProvider({
      clientId: process.env.GITHUB_ID,
      clientSecret: process.env.GITHUB_SECRET,
    }),
    // ...add more providers here
  ],
};
```

Reply:
the if statement checks if we are in the vercel preview environment or not. Because we can't use gitHub provider in vercel preview mode, I built custom provider with dummy credentials.

Here is what Coach Ralf Posted in our Slack Channel 

"Hallo zusammen, hier noch ein Nachtrag bezüglich NextAuth und den Preview-Deployments, die Vercel für unsere Pull-Requests erstellt: Da die OAuth-Provider (Github, Google, etc.) in diesen Environments nicht funktionieren, kann man speziell für diese Deployments einen Dummy-Login mit dem Credentials Provider konfigurieren (z.B. mit username="test" und password="test") So kann man auch im Preview-Deployment die Features für eingeloggte User testen. Vielen Dank an 
[@Omer](https://neuefische-students.slack.com/team/U03PMSL79BL)
 für die Frage im Discussions-Board, hier der Link zu den Details:"

Reply:
Thanks for the clarification. Haven't checked the proper logic out since I've not been working with a login myself.
Will probably need this part later on as well 😊